crates/lang/macro/tests/compile_tests.rs

```diff
@@ -23,7 +23,17 @@ fn compile_tests() {
     t.pass("tests/ui/pass/05-erc721-contract.rs");
     t.pass("tests/ui/pass/06-non-ink-items.rs");
     t.pass("tests/ui/pass/07-flipper-as-dependency.rs");
-    t.pass("tests/ui/pass/08-flipper-as-dependency-trait.rs");
+    if option_env!("INK_COVERAGE_REPORTING") != Some("true") {
+        // The cross-calling implementation for traits provides
+        // an invalid implementation for non-valid message calls
+        // (e.g. cross-calling a `mut` message from a non-`mut` message).
+        // So calling those will result in a compiler or linker error.
+        //
+        // The coverage reporting CI stage though also links dead code,
+        // hence resulting in this invalid implementation being linked
+        // and thus a linker error.
+        t.pass("tests/ui/pass/08-flipper-as-dependency-trait.rs");
+    }
```

Comment:
If ink! could find out when it is compiled under coverage reporting then ink! could avoid producing those enforced linker errors. Maybe that's a better solution?

Reply:
I covered this in the description of this PR:

>Another way of implementing this would be to place an early return in the methods which insert `__ink_enforce_error` in [`cross_calling.rs`](https://github.com/paritytech/ink/blob/master/crates/lang/codegen/src/generator/cross_calling.rs) (in case `INK_COVERAGE_REPORTING` is set to `true`). I've decided against this, since those code paths would then be excluded for all tests which are run in our `codecov` CI stage, making it impossible to improve coverage for them.

tl;dr is that by excluding those code paths I think we skew our code coverage results without any possibility of improving on the left out code branches.